### PR TITLE
update e2e-test to support windows

### PR DESCRIPTION
**Pull request by monianshouhou**
#### What type of PR is this?
/kind cleanup
/kind failing-test

#### What this PR does / why we need it:
"should be able to pull private images from acr without docker secrets set explicitly" test will fail in Windows because linux testing environment always pull linux images which can't be pulled successfully in Windows.
Other tests can successfully run in Windows.
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by monianshouhou** [comment]
@lzhecheng  PTAL

File: tests/e2e/auth/cred.go [hunk 1]
```diff
@@ -99,6 +99,21 @@ func createPodPullingFromACR(registryName, image, tag string) (result *v1.Pod) {
 					Image: registryName + ".azurecr.io/" + image + ":" + tag,
 				},
 			},
+			NodeSelector: map[string]string{
+				"kubernetes.io/os": "linux",
+			},
+			Tolerations: []v1.Toleration{
+				{
+					Key:      "node-role.kubernetes.io/control-plane",
```
Comment (lzhecheng) [inline tests/e2e/auth/cred.go:107]: There's defined var for this.
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/21a6572fca7bc25aa6a3f0672a28fc31709030d1/tests/e2e/utils/node_utils.go#L36

File: tests/e2e/auth/cred.go [hunk 2]
```diff
@@ -99,6 +99,21 @@ func createPodPullingFromACR(registryName, image, tag string) (result *v1.Pod) {
...
+					Operator: v1.TolerationOpExists,
+					Effect:   v1.TaintEffectNoSchedule,
+				},
+				{
+					Key:      "node-role.kubernetes.io/master",
```
Comment (lzhecheng) [inline tests/e2e/auth/cred.go:112]: ditto
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/21a6572fca7bc25aa6a3f0672a28fc31709030d1/tests/e2e/utils/node_utils.go#L35

File: tests/e2e/auth/cred.go (hunk 1 above)
Comment (monianshouhou) [reply tests/e2e/auth/cred.go:107]: OK! I will fix that.

File: tests/e2e/auth/cred.go [hunk 3]
```diff
@@ -99,6 +99,21 @@ func createPodPullingFromACR(registryName, image, tag string) (result *v1.Pod) {
 					Image: registryName + ".azurecr.io/" + image + ":" + tag,
 				},
 			},
+			NodeSelector: map[string]string{
+				"kubernetes.io/os": "linux",
```
Comment (lzhecheng) [inline tests/e2e/auth/cred.go:103]: https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/tests/e2e/utils/node_utils.go#L38

File: tests/e2e/auth/cred.go (hunk 1 above)
Comment (monianshouhou) [reply tests/e2e/auth/cred.go]: I have fixed that. PTAL.

**Pull request merged**